python/lancedb/db.py

```diff
@@ -287,6 +287,7 @@ def table_names(self) -> list[str]:
             for file_info in paths
             if file_info.extension == "lance"
         ]
+        tables.sort()
```

Comment:
Let's mention in the method description that table names are returned in sorted order.  Also, this is technically a breaking change.  Let's [label it as such](https://www.conventionalcommits.org/en/v1.0.0/).

Reply:
done & done. Thanks @westonpace